torchvision/models/_api.py

```diff
@@ -107,10 +110,27 @@ def get_weight(name: str) -> WeightsEnum:
     return weights_enum.from_str(value_name)
 
 
+W = TypeVar("W", bound=Type[WeightsEnum])
+
+
+def get_model_weight(model: Union[Callable, str]) -> W:
```

Comment:
It feels a bit unintuitive to me that this returns metadata about the weights instead of the actual model parameters. it's probably too late to change the enums, but I wonder if we should have named them WeightsMetadata, or if this be any clearer as something like `get_model_weight_enum`?

Reply:
Thanks @nairbv for the feedback.

I agree that the names can be confusing but thankfully these are just the base classes. The concrete user-facing enums look like `ResNet50_Weights`, so hopefully this doesn't lead to confusion. I can make the change to enum if you feel it's going to be clearer, just note that the word `enum` is not actually visible to the users. WDYT?

Reply:
+1 to the suggestion of renaming to `get_model_weights`

Reply:
I had an offline chat with @nairbv and his main concern is that the current naming doesn't make it clear whether a single Weight entry is returned versus the whole class. I think thus renaming this to `get_model_weights_enum()` makes sense. I'll update the PR accordingly.

Reply:
@jdsgomes I missed your reply. Any thoughts on Brian's proposal? cc @NicolasHug

Reply:
No strong opinion. The minor issue with `get_model_weights_enum` is that it exposes an implementation detail (the `enum` part).

Like @jdsgomes , I think `get_model_weights` makes more sense than `get_model_weight` since all of these objects end with `Weights` ("s" at the end).

Reply:
Thanks everyone for your input. I think I'll move forward with `get_model_weights()` for now. I'm not going to merge the PR but instead leave this be the live RFC doc. If there are strong concerns over the naming, we can address them on the actual PR that lands the code (coming up). If you have more concerns, please keep commenting on this doc.